scripts/lib/build-utils.ts

```diff
@@ -295,6 +312,112 @@ export default {
     });
   },
 
+  async buildExtensionProxyImage(): Promise<void> {
+    const workDir = await fs.promises.mkdtemp(path.join(os.tmpdir(), 'rd-build-rdx-pf-'));
+
+    try {
+      const executablePath = path.join(workDir, 'extension-proxy');
+      const layerPath = path.join(workDir, 'layer.tar');
+      const imagePath = path.join(this.rootDir, 'resources', 'rdx-proxy.tgz');
+
+      console.log('Building RDX proxying image...');
+
+      // Build the golang executable
+      await this.spawn('go', 'build', '-ldflags', '-s -w', '-o', executablePath, '.', {
+        cwd: path.join(this.rootDir, 'src', 'go', 'extension-proxy'),
+        env: { ...process.env, GOOS: 'linux' },
+      });
+
+      // Build the layer tarball
+      // tar streams don't implement piping to multiple writers, and stream.Duplex
+      // can't deal with it either; so we need to fully writ out the file, then
+      // calculate the hash as a separate step.
+      const layer = tar.pack();
```

Comment:
I don't see this tar API at https://www.npmjs.com/package/tar -- could you add a link pointing to the API you're using in this module?

Reply:
I'm using https://www.npmjs.com/package/tar-stream — see the top of the file where it's imported?

Reply:
Sorry, I searched npm before checking the import statements